utils_cv/detection/dataset.py

```diff
@@ -20,12 +20,48 @@
 from .bbox import AnnotationBbox
 from .mask import binarise_mask
 from .references.utils import collate_fn
-from .references.transforms import RandomHorizontalFlip, Compose, ToTensor
+from .references.transforms import Compose, ToTensor
 from ..common.gpu import db_num_workers
 
 Trans = Callable[[object, dict], Tuple[object, dict]]
 
 
+def _flip_keypoints(kps, width, hflip_inds):
+    """ Variation of `references.transforms._flip_coco_person_keypoints` with additional
+    hflip_inds. """
+    flipped_data = kps[:, hflip_inds]
+    flipped_data[..., 0] = width - flipped_data[..., 0]
+    # Maintain COCO convention that if visibility == 0, then x, y = 0
+    inds = flipped_data[..., 2] == 0
+    flipped_data[inds] = 0
+    return flipped_data
+
+
+class RandomHorizontalFlip(object):
+    """ Variation of `references.transforms.RandomHorizontalFlip` to make sure flipping
+    works on custom keypoints. """
+
+    def __init__(self, prob):
+        self.prob = prob
+
+    def __call__(self, image, target):
```

Comment:
[minor] image should probably be called "im" for consistency

Reply:
Done